### PR TITLE
feat: PP-11306 - Adds multiple options for autopick allowing for multiple groups per instruction

### PR DESCRIPTION
**Pull request by AlexanderNull**
Enables providing multiple groups of picks per single autopick instruction

### CODE REVIEW

File: autoprotocol/types/protocol.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from dataclasses import dataclass
```
Comment (ericschade) [inline autoprotocol/types/protocol.py:1]: this is the additional import you mentioned which is the tiebreaker in terms of which option you prefer?

Comment (ericschade) [reply autoprotocol/types/protocol.py:1]: nvm. when you say consumers youre talking about ap writers, yeah? the import would be `from protocol.types import pickparams`

**Comment by ericschade** [comment]
im also pretty neutral between the two options but probably lean slightly towards the param class because i like the idea of being able to 

```
myparams = AutopickParams(source = s, dest = d, min_abort = m_a)
```
or 
```
myparams = AutopickParams(dest = d, source = s, min_abort = m_a)
```
and not have to remember that tuple[0] is the source while tuple[1] is the dest(either order would pass the type hint since both are of type `WellParam`)

File: autoprotocol/protocol.py [hunk 2]
```diff
@@ -5081,6 +5083,97 @@ def autopick(
             Source wells are not all from the same container
 
         """
+        groups = []
+        for group in pick_groups:
+            sources, dests, min_abort = pick_groups
```
Comment (AlexanderNull) [inline autoprotocol/protocol.py:5088]: ... = group

File: test/protocol_test.py [hunk 3]
```diff
@@ -1691,14 +1732,46 @@ def test_autopick(self):
...
+            "agar_plate",
+            {"reserve": "ki17reefwqq3sq", "discard": True},
+            Container(None, p.container_type("6-flat"), name="agar_plate"),
         )
 
+        agar_plate = Container(None, p.container_type("6-flat"), name="agar_plate")
+        source1 = [agar_plate.well(0), agar_plate.well(1)]
+        dest1 = [dest_plate.well(1)] * 4
+        pick_group = AutopickGroupClass(source1, dest1)
+        pick_group = ([agar_plate.well(0), agar_plate.well(1)], [dest_plate.well(1)] * 4, 0)
```
Comment (AlexanderNull) [inline test/protocol_test.py:1771]: remove second duplicate line

File: autoprotocol/types/protocol.py (hunk 1 above)
Comment (AlexanderNull) [reply autoprotocol/types/protocol.py:1]: an example of it is down in the test that uses autopick2

File: autoprotocol/protocol.py [hunk 4]
```diff
@@ -5081,6 +5082,27 @@ def autopick(
             Source wells are not all from the same container
 
         """
+        groups = []
```
Comment (EribertoLopez) [inline autoprotocol/protocol.py:5085]: It may be good to have a check that checks that the group is an instance of the AutopickGroup class.

**Review by aaronkarlsberg1 — DISMISSED** [review]
lgtm, stronger typing for the win! Echoing Eddie's suggestion of runtime check that the group is of type AutopickGroup.

**Review by EribertoLopez — DISMISSED** [review]

**Review by EribertoLopez — APPROVED** [review]

**Pull request merged**